### DISCUSSION TITLE
What would you like to see in Sentinel Go 2.0?

### DISCUSSION THREAD

**Discussion by sczyh30** [alibaba/sentinel-golang#399] [Ideas]
What would you like to see in Sentinel Go 2.0 (regarding cloud-native, adaptive flow control, distributed rate limiting, or any other aspects)? Any ideas are welcomed!

**Comment by sczyh30** [comment]
Some ideas of my own:

- A [cluster policy brain](https://github.com/alibaba/sentinel-golang/issues/401) for cluster-level policy enforcement (e.g. rate-limiting, access control, service outlier ejection, service discovery adjustment). This can be a fundamental cloud-native component (NOT just a lib).
- [Eventing](https://github.com/alibaba/sentinel-golang/issues/402) (everything can be regarded as a kind of event)
- [Adaptive control abstraction](https://github.com/alibaba/Sentinel/issues/1889)

Any other ideas? @louyuting @sanxun0325 @luckyxiaoqiang @jasonjoo2010 @cdfive

**Comment by jasonjoo2010** [comment]
The distributed flow control hasn't been aligned to sentinel for a long time which is also requested several times. I saw the tech design has been completed in the issue list and it was labeled in the milestone of 1.2 . So will it be available in 1.2 or 2.0?

And furthermore, it would be better if the distributed flow control could be self coordinated.
